### DISCUSSION TITLE
No issue, more of a question.

### DISCUSSION THREAD

**Discussion by bjornenvall** [gekkekoe/esphome-ecodan-hp#20] [General]
Do you know if it’s possible to have the WiFi MELCloud thing connected at the same time?

**Comment by gekkekoe** [comment]
I have beta software that supports that, but it will not work on the plain atom s3 (I used that setup to reverse engineer cmds).
You need to solder to get it working. I have it locally by splitting the 12v (procon won't power on with 5v), and wire another set of rx/tx pins to the esp and proxy the data between heatpump and wifi adapter/procon, but that would make this project hard to use, and that was not the goal.

I also see no real use for melcloud when you have a local home assistant (you can access all data via HA via vpn or HA cloud).

**Comment by bjornenvall** [comment]
**thank you so much for this!** close to plug n play program. m5stack s3 lite.

**Comment by bjornenvall** [comment]
i have a daikin ac that I'm trying to do the same thing with. can't u make this too? I'm struggling with this: https://github.com/joshbenner/esphome-daikin-s21

**Reply by dwmw2** [reply]
For Daikin see https://github.com/revk/ESP32-Faikin which isn't esphome but it does feed HA in MQTT. The hardware is nice too; I've got one (from [Amazon](https://www.amazon.co.uk/dp/B0C2ZYXNYQ)) to hook up to my Ecodan units.

**Comment by gekkekoe** [comment]
Sorry don't have a unit, so unable to test, but it does look similar

**Comment by loopyengineeringco** [comment]
In the UK, there was an financial incentive called MMSP which meant the energy regulator Ofgem get the regular data from MelCloud for statistical purposes - and we got a nice little wedge of cash :-)
So that would be a use case where we need to retain the MelCloud comms to keep the data flowing to the government but also have local control/polling. As far as I could see there's no way to control cooling mode on/off from the melcloud app, it's awful in general - having it in esphome/HA/mqtt is a massive upgrade.

What did you use to proxy the data, an esp8266/32 and a second serial? Would you be able to post the code somewhere?
Thank you for your time on this project as well, it's fantastic and super easy for those who are able to use it!

**Comment by gekkekoe** [comment]
I use the standard esp32 and 2 serial ports (basically forwarding data from slave to master). you probably need to use the 12v on the cn105 to power the wifi adapter, the 5v could not power the procon, so I would expect the same for the wifi adapter. The code is really beta and hacky since it was never intended to run in 'production'. If you're able to adjust it, then you can write it yourself ;-) Have a look at the panasonic heishamon project.

If they read data from melcloud, isn't it easier to post the data via api calls directy from HA to melcloud?

**Reply by loopyengineeringco** [reply]
No worries, thanks for the info. Will check out the Panasonic project.
Could be easier, but I don't really know what I'm doing when it comes to authenticating and posting to the API, I'm better with the hardware 🤣

**Reply by gekkekoe** [reply]
heishamon also uses 2 sets of serial ports to proxy. you can look at how they implemented it, hardware is a very small portion of the solution

**Comment by gekkekoe** [comment]
it's coming! :)

We (me and @dwmw2 ) have proxy mode working with the heishamon board (but should work on all esp32 with some levelshifters). We still need to do some testing, but you can run the esp and the melcloud wifi adapter together. You can even change setting from the esp. It needs a bit more testing and will be merged to main after testing.
more info about heishamon: https://www.tindie.com/products/thehognl/heishamon-communication-pcb/
current experimental impl: https://github.com/gekkekoe/esphome-ecodan-hp/tree/proxy

**Reply by dwmw2** [reply]
Don't forget we need cables with the right pinout, and to add a pull-up 10kΩ resistor on the RX line from the heat pump, for the Heishamon.
![image](https://github.com/user-attachments/assets/3a56fdea-e989-44a5-9bdc-591518f09253)

**Reply by dwmw2** [reply]
![image](https://github.com/user-attachments/assets/f5f0f805-18dc-4844-9e36-f74b338067a6)

**Reply by F1p** [reply]
> Don't forget we need cables with the right pinout, and to add a pull-up 10kΩ resistor on the RX line from the heat pump, for the Heishamon. 

Was it not possible to get the esp internal 45k pullup to work?

**Reply by dwmw2** [reply]
The board needs to have a MOSFET for input voltage protection (in *some* models the TX from the heat pump is at 5v), so the pull-up on the chip itself doesn't help.

Of course, since the TX from the heat pump *isn't* actually at 5v on these Ecodan models, we could bypass the voltage protection and just wire it directly to any GPIO without worrying... and then yes, you could use the internal pullup.

**Reply by gekkekoe** [reply]
It  has been merged, I will do a write up later, but for now the schematics are in the img folder. For both heishamon and atoms3 (applies to generic esp32)
